xapian-applications/omega/omega.cc

```diff
@@ -447,17 +467,21 @@ try {
     // Percentage relevance cut-off
     val = cgi_params.find("THRESHOLD");
     if (val != cgi_params.end()) {
-	threshold = atoi(val->second.c_str());
-	if (threshold < 0) threshold = 0;
-	if (threshold > 100) threshold = 100;
+	unsigned int temp;
+	if (!parse_unsigned(val->second.c_str(),temp) || temp > 100) {
+	    throw "THRESHOLD parameter must be in the range 0-100\n";
+	}
+	threshold = temp;
     }
 
     // collapsing
     val = cgi_params.find("COLLAPSE");
     if (val != cgi_params.end()) {
 	const string & v = val->second;
 	if (!v.empty()) {
-	    collapse_key = atoi(v.c_str());
+	    if (!parse_unsigned(val->second.c_str(),collapse_key)) {
```

Comment:
And another - just fix all instances of this please!

Reply:
thanks i will fix that.